File: Python/8py 4/xor.py
"""
|--------------------------------------------------------------------------
| Exclusive "or" (xor) Logical Operator
|--------------------------------------------------------------------------
|
"""

def xor(a, b):
   return a^b


from operator import xor
xor = int.__xor__


def xor(a,b):
   return a != b


def xor(a,b):
   return a is not b


print(xor(False, False), False)
print(xor(True, False), True)
print(xor(False, True), True)
print(xor(True, True), False)

File: JavaScript/8js 3/xor.js
/*
|--------------------------------------------------------------------------
| Exclusive "or" (xor) Logical Operator
|--------------------------------------------------------------------------
|
*/

function xor(a, b) {
   return (a ^ b) == 1 ? true : false;
}


function xor(a, b) {
   return a != b;
}


function xor(a, b) {
   return !!(a ^ b);
}


function xor(a, b) {
   return (a || b) && !(a && b);
}


console.log(xor(false, false), false);
console.log(xor(true, false), true);
console.log(xor(false, true), true);
console.log(xor(true, true), false);
console.log(xor(true, true), true);